### DISCUSSION TITLE
Icons on TreeView

### DISCUSSION THREAD

**Discussion by DorianGrayCodes** [cogentcore/core#433] [Q&A] [answered]
Hello,

How do I set icons for specific children in a TreeView? I tried setting the Icon field for TreeView and it only changed the icon for the root node, but I would like to set icons on a child by child basis.

Thanks,
Daniel

**Comment by rcoreilly** [comment, marked as answer]
It should definitely work to set per node -- FileTreeView does that for example, to indicate the file type.

**Reply by DorianGrayCodes** [reply]
I forgot to respond to this, but I realized the way to do this was to make every single node its own FileTreeView, and then I could set the Icon for it

**Comment by kkoreilly** [comment]
This can also now be done via an `OnWidgetAdded` that customizes all of the tree view children. Also, we now support `IconOpen`, `IconClosed`, and `IconLeaf`.
